std-bits/table/src/main/java/org/enso/table/data/column/operation/unary/DatePartOperation.java

```diff
@@ -55,26 +56,38 @@ public class DatePartOperation extends AbstractUnaryLongOperation {
   public static final UnaryOperation MILLISECOND_INSTANCE =
       new DatePartOperation(MILLISECOND, ChronoField.MILLI_OF_SECOND, true);
 
+  protected final String name;
   protected final TemporalField field;
   protected final boolean timeField;
 
   protected DatePartOperation(String name, TemporalField field, boolean timeField) {
-    super(name, true, IntegerType.INT_64);
+    this.name = name;
     this.field = field;
     this.timeField = timeField;
   }
 
+  @Override
+  public String getName() {
+    return name;
+  }
+
   @Override
   public boolean canApply(ColumnStorage<?> storage) {
     return timeField ? storage.getType().hasTime() : storage.getType().hasDate();
   }
 
   @Override
-  protected void applyObjectRow(
-      Object value, BuilderForLong builder, MapOperationProblemAggregator problemAggregator) {
+  public ColumnStorage<?> apply(
+      ColumnStorage<?> storage, MapOperationProblemAggregator problemAggregator) {
+    return StorageIterators.buildOverStorage(
+        storage,
+        Builder.getForLong(IntegerType.INT_64, storage.getSize(), problemAggregator),
+        (builder, index, value) -> builder.appendLong(applyObjectRow(index, value)));
+  }
+
+  protected long applyObjectRow(long index, Object value) {
```

Comment:
Why the index argument? Struggling to find where we ever use it/

Reply:
For the framework seemed good to have it as would allow zip operations etc in due cource.